src/main/lock.ts

```diff
@@ -0,0 +1,44 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as debug_ from "debug";
+import { app } from "electron";
+import { container } from "readium-desktop/main/di";
+import { WinRegistry } from "readium-desktop/main/services/win-registry";
+
+import { cli } from "./cli/process";
+
+// Logger
+const debug = debug_("readium-desktop:main:lock");
+
+export function lockInstance() {
+    const gotTheLock = app.requestSingleInstanceLock();
+
+    if (gotTheLock) {
+        app.on("second-instance", (_e, commandLine, _workingDir) => {
+            // Someone tried to run a second instance, we should focus our window.
+            debug("comandLine", commandLine, _workingDir);
+
+            const winRegistry = container.get("win-registry") as WinRegistry;
+            const win = winRegistry.getWindow(1);
+            if (win && win.win) {
+                if (win.win.isMinimized()) {
+                    win.win.restore();
+                }
+                win.win.focus();
+            }
+
+            // execute command line from second instance
+            // when the command line doesn't used electron: execute and exit in second instance process
+            // when the command has needed to open win electron: execute with below cli function
+            // the mainFct is disallow to avoid to generate new mainWindow
+            // remove all -- option arg because isn't handle in ready state app
+            cli(() => ({}), commandLine.filter((arg) => !arg.startsWith("--")));
```

Comment:
This will be impacted by https://github.com/readium/readium-desktop/pull/607/files